src/lua/funcs/mod.rs

```diff
@@ -0,0 +1,33 @@
+//! Rust functions for lua libraries
+
+use hlua;
+use hlua::{Lua, LuaTable, Function};
+
+// Prevent functions declared here from not being registered
+#[forbid(dead_code)]
+#[forbid(unused_variables)]
+
+mod input;
+
+/// Register all the Rust functions for the lua libraries
+pub fn register_libraries(lua: &mut Lua) {
+
+    // A macro for adding lua functions... could be a method but whatever
+    macro_rules! lua_function {
```

Comment:
If this could be a method, it probably should be. I don't know Rust's stance on that grey line for functions that could be functions or macros, but to take a leaf out of Scheme/Lisp's book I'd say if there is something that could be a function that is instead written as a macro, you should probably write it as a function. Macros are hard to understand, and add an unnecessary level of complexity in certain cases.

Now I don't know Rust's syntax for macros, but from what I can see is that it makes a new function then automatically binds it with lua.set. Can that be a function pointer/lambda of some kind? Because it looks like you pass the function at `hlua::function0($name);`, so why do we need to have it defined? And if so, we could have it be returned as a lambda/function pointer. If you have good reasons to keep it in I'm all for hearing them, but I think this is something important to consider.